crates/anvil-polkadot/src/api_server/server.rs

```diff
@@ -167,10 +344,192 @@ impl ApiServer {
+            .await
+            .map_err(Error::Revive)?;
+        let code = self
+            .eth_rpc_client
+            .runtime_api(hash)
+            .code(ReviveAddress::from(address).inner())
+            .await
+            .map_err(Error::Revive)?;
+        Ok(code.into())
+    }
+
+    async fn get_block_by_hash(
+        &self,
+        block_hash: B256,
+        hydrated_transactions: bool,
+    ) -> Result<Option<Block>> {
+        let Some(block) = self
+            .eth_rpc_client
+            .block_by_hash(&H256::from_slice(block_hash.as_slice()))
+            .await
+            .map_err(Error::Revive)?
+        else {
+            return Ok(None);
+        };
+        let block = self.eth_rpc_client.evm_block(block, hydrated_transactions).await;
+        Ok(Some(block))
+    }
+
+    async fn estimate_gas(
+        &self,
+        request: WithOtherFields<TransactionRequest>,
+        block: Option<alloy_rpc_types::BlockId>,
+    ) -> Result<sp_core::U256> {
+        node_info!("eth_estimateGas");
+
+        let hash = self
+            .eth_rpc_client
+            .block_hash_for_tag(ReviveBlockId::from(block).inner())
+            .await
+            .map_err(Error::Revive)?;
+        let runtime_api = self.eth_rpc_client.runtime_api(hash);
+        let dry_run = runtime_api
+            .dry_run(convert_to_generic_transaction(request.into_inner()))
+            .await
+            .map_err(Error::Revive)?;
+        Ok(dry_run.eth_gas)
+    }
+
+    async fn gas_price(&self) -> Result<sp_core::U256> {
+        let hash = self
+            .eth_rpc_client
+            .block_hash_for_tag(BlockTag::Latest.into())
+            .await
+            .map_err(Error::Revive)?;
+
+        let runtime_api = self.eth_rpc_client.runtime_api(hash);
+        runtime_api.gas_price().await.map_err(Error::Revive)
+    }
+
+    pub async fn get_transaction_count(
+        &self,
+        address: H160,
+        block: BlockNumberOrTagOrHash,
+    ) -> Result<sp_core::U256> {
+        let hash = self.eth_rpc_client.block_hash_for_tag(block).await.map_err(Error::Revive)?;
+        let runtime_api = self.eth_rpc_client.runtime_api(hash);
+        let nonce = runtime_api.nonce(address).await.map_err(Error::Revive)?;
+        Ok(nonce)
+    }
+
+    async fn send_raw_transaction(&self, transaction: Bytes) -> Result<H256> {
+        let hash = H256(keccak_256(&transaction.0));
+        let call = subxt_client::tx().revive().eth_transact(transaction.0);
+        self.eth_rpc_client
+            .submit(call)
+            .await
+            .map_err(|err| {
+                node_info!("submit call failed: {err:?}");
+                err
+            })
+            .map_err(Error::Revive)?;
+
+        node_info!("send_raw_transaction hash: {hash:?}");
+        Ok(hash)
+    }
+
+    pub(crate) async fn send_transaction(
+        &self,
+        transaction_req: WithOtherFields<TransactionRequest>,
+    ) -> Result<H256> {
```

Comment:
Also would be great to add a `node_info!("eth_sendTransaction");` log at the beginning of the method call. Similarly to the other RPCs.